python/ql/test/experimental/dataflow/use-use-flow/read_explosion.py

```diff
@@ -0,0 +1,36 @@
+# This test file is inspired by
+# `csharp/ql/test/library-tests/dataflow/local/UseUseExplosion.cs`
+# but with `n=3` kept small, since we do have the explosion.
+
+cond = ...
+
+# global variables are slightly special,
+# so we go into a function scope
+def scope():
+    x = 0
+
+    if(cond > 3):
```

Comment:
You could also add a test for normal phi nodes, i.e. something like
```python
    if(cond > 3):
      if(cond > 2):
        if(cond > 1):
           pass
        else:
          x = 1
      else:
        x = 2
    else:
      x = 3

    if(cond > 3):
      if(cond > 2):
        if(cond > 1):
           pass
        else:
          use(x)
      else:
        use(x)
    else:
      use(x)
```

Reply:
I put that in https://github.com/github/codeql/pull/14861. (Looks like I came to the exact same example :-))